validations/transfer.test.ts

```diff
@@ -0,0 +1,136 @@
+import { expect, use } from 'chai'
+import { Wallet, ethers } from 'ethers'
+import path from 'path'
+import { randomUUID } from 'crypto'
+import { AddKycParams, FiatConnectClient } from '@fiatconnect/fiatconnect-sdk'
+import {
+  KycSchema,
+  TransferStatus,
+  Network,
+} from '@fiatconnect/fiatconnect-types'
+import { config } from '../src/config'
+import { chaiPlugin } from 'api-contract-validator'
+import { MOCK_QUOTE } from '../src/mock-data/quote'
+import { MOCK_KYC } from '../src/mock-data/kyc'
+import { MOCK_FIAT_ACCOUNTS } from '../src/mock-data/fiat-account'
+
+const apiDefinitionsPath = path.join(config.openapiSpec)
+use(chaiPlugin({ apiDefinitionsPath }))
+
+describe('/transfer', () => {
+  const mockAccountData =
+    MOCK_FIAT_ACCOUNTS[
+    config.fiatAccountMock as keyof typeof MOCK_FIAT_ACCOUNTS
+  ]
+
+  const mockKYCInfo: AddKycParams<KycSchema> =
+    MOCK_KYC[config.kycMock as keyof typeof MOCK_KYC]
+
+  describe('/in', () => {
+    const wallet = Wallet.createRandom()
+
+    const fiatConnectClient = new FiatConnectClient(
+      {
+        baseUrl: config.baseUrl,
+        network: Network.Alfajores,
+        accountAddress: wallet.address,
+      },
+      (message: string) => wallet.signMessage(message),
+    )
+
+    const quoteInParams = {
+      ...MOCK_QUOTE[config.quoteInMock],
+      address: wallet.address,
+    }
+
+    it('able to transfer fiat in for crypto', async () => {
+      const loginResult = await fiatConnectClient.login()
+      expect(loginResult.isOk).to.be.ok
+
+      const quoteInResponse = await fiatConnectClient.createQuoteIn(
+        quoteInParams,
+      )
+      expect(quoteInResponse.isOk).to.be.ok
+      const quoteId = quoteInResponse.unwrap().quote.quoteId
+
+      const addKycResult = await fiatConnectClient.addKyc(mockKYCInfo)
+      expect(addKycResult.isOk).to.be.true
+
+      const addAccountResult = await fiatConnectClient.addFiatAccount(
+        mockAccountData,
+      )
+      expect(addAccountResult.isOk).to.be.true
+      const fiatAccountId = addAccountResult.unwrap().fiatAccountId
+
+      const transferInParams = {
```

Comment:
Since we use idemopotency keys on the transfer endpoints (currently, _only_ the transfer endpoints), it might be a good idea to perform an identical transfer with the same idempotency keys/params and ensure that the response is exactly identical to the first transfer.

Reply:
Yeah that makes sense. If we are using the same wallet address at least for the transfer out test and possibly for all tests, wouldn't that mean there's some risk of unexpected duplicate idempotency key errors depending on the provider's [idempotency key policy](https://datatracker.ietf.org/doc/html/draft-ietf-httpapi-idempotency-key-header-00#section-2.3)?

Reply:
Hrm, not sure I understand the concern here. If we use a random idemopotency key for each request, we shouldn't ever run into a case where we re-use a key for actual different requests, right?

Reply:
Ah I see, I think I misunderstood what you were suggesting in your first comment. That's a good test, will add it in.